numpy/core/tests/test_regression.py

```diff
@@ -2115,6 +2115,10 @@ def __array__(self,*args,**kwargs):
         for f in [op.lt, op.le, op.gt, op.ge]:
             if sys.version_info[0] >= 3:
                 assert_raises(TypeError, f, lhs, rhs)
+            elif sys.version_info[0] < 3 and sys.py3kwarning:
+                warnings.filterwarnings('error',category=DeprecationWarning)
+                assert_raises(DeprecationWarning, f, lhs, rhs)
+                warnings.filterwarnings('ignore',category=DeprecationWarning)
```

Comment:
Since the tests run anyhow, I am OK with skipping when its easier. Though suppression is cleaner and seems preferable to me if it is straight forward.